keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -897,14 +917,15 @@ class SqlDeliveryConfigRepository(
   }
 
   // todo eb: add link to artifact https://github.com/spinnaker/keel/issues/1270
-  override fun getQueuedConstraintApprovals(deliveryConfigName: String, environmentName: String): Set<String> {
+  override fun getQueuedConstraintApprovals(deliveryConfigName: String, environmentName: String, artifactType: String): Set<String> {
     val environmentUID = environmentUidByName(deliveryConfigName, environmentName)
       ?: return emptySet()
 
     return sqlRetry.withRetry(READ) {
       jooq.select(ENVIRONMENT_ARTIFACT_QUEUED_APPROVAL.ARTIFACT_VERSION)
         .from(ENVIRONMENT_ARTIFACT_QUEUED_APPROVAL)
         .where(ENVIRONMENT_ARTIFACT_QUEUED_APPROVAL.ENVIRONMENT_UID.eq(environmentUID))
+        .and(ENVIRONMENT_ARTIFACT_QUEUED_APPROVAL.ARTIFACT_TYPE.eq(artifactType))
```

Comment:
adding this here means that when you roll this out we won't consider older versions with `artifactType == null` (all the old data). It's probably worth allowing null here as well